### PR TITLE
feat: e2e cas_analyst

### PR DESCRIPTION
**Pull request by shon-button**
Addresses:
 [845](https://github.com/bcgov/cas-registration/issues/845)
[1295](https://github.com/bcgov/cas-registration/issues/1295)

## 🚀 Impact:
   -  CI `Test Registration App / 📊 e2e report artifact` is triggered on e2e-test failure and skipped on e2e-test success
   -  CI `Test Registration App / 📊 e2e report artifact` merging only `blob-report-chromium` as workaround to occaisiona error "invalid signature" due to corrupt file?
   -  Implements `cas_analyst` isolated workflow tests

##  🔬 Testing:

CI e2e test
Validate [e2e test](https://github.com/bcgov/cas-registration/actions/runs/8631313923) status is  `pass` 

image

### CODE REVIEW

**Review by BCerki — COMMENTED** [review]
Great work on this! And we're almost done with e2e testing, woooo! 

It's been a while since I looked at the e2e tests so I delved into the POMs. Some of my suggestions are about code written before this PR, so if you feel addressing them here is scope creep, feel free to make tech tickets instead.

A few overall notes:
- Sonarcloud fails are all about duplication so I think we can silence them
- We're missing happo screenshots in the cas_analyst test (and in all the cas user tests and the new user test, but that can be tech debt since this card is about cas_analyst)
- We may need a new ticket about accessibility testing in this test depending on when @marcellmueller 's PR goes on

File: client/e2e/workflows/idir/cas_admin.spec.ts [hunk 1]
```diff
@@ -32,82 +32,107 @@ test.afterAll(async () => {
...
+      // 🔍 Assert cas_admin is able to click "View Details" on see detailed info related Pending
+      await operatorsPage.formHasExpectedUX(UserOperatorStatus.PENDING);
+    });
+
+    test("Test details form by workflow", async ({ page }) => {
+      const operatorsPage = new OperatorsPOM(page);
+      // 🛸 Navigate to operators page
+      operatorsPage.route();
+      // 🔍 Assert cas_admin workflow New Operator, Pending: Reject
+      await operatorsPage.formHasExpectedWorkflow(UserRole.CAS_ADMIN, 1);
```
[the remark this answers is not in the document]
Comment (shon-button) [reply client/e2e/workflows/idir/cas_admin.spec.ts:74]: I felt caseIndex implied the "this is the number for this case of testing..." but, I will defer to your linguistic prowess.

File: client/e2e/poms/operations.ts [hunk 2]
```diff
@@ -278,7 +279,7 @@ export class OperationsPOM {
               this.alertApproved,
             );
             // FIXME FOR CI
-            //   await expect(this.messageOperationApproved).toBeVisible();
+            // await expect(this.messageOperationApproved).toBeVisible();
```
[the remark this answers is not in the document]
Comment (shon-button) [reply client/e2e/poms/operations.ts:282]: Being addressed [here](https://github.com/bcgov/cas-registration/issues/1338)

File: client/e2e/workflows/idir/cas_admin.spec.ts [hunk 3]
```diff
@@ -32,82 +32,107 @@ test.afterAll(async () => {
...
+      operatorsPage.route();
+      // 🔍 Assert cas_admin is able to click "View Details" on see detailed info related Declined
+      await operatorsPage.formHasExpectedUX(UserOperatorStatus.DECLINED);
+      // 🔍 Assert cas_admin is able to click "View Details" on see detailed info related Approved
+      await operatorsPage.formHasExpectedUX(UserOperatorStatus.APPROVED);
+      // 🔍 Assert cas_admin is able to click "View Details" on see detailed info related Pending
+      await operatorsPage.formHasExpectedUX(UserOperatorStatus.PENDING);
+    });
+
+    test("Test details form by workflow", async ({ page }) => {
```
[the remark this answers is not in the document]
Comment (shon-button) [reply client/e2e/workflows/idir/cas_admin.spec.ts:69]: Do the assert documentations not fulfil spec documentation sufficiently?

```
// 🔍 Assert cas_analyst workflow Pending, Request Changes, Undo (Request Changes), Approve
// 🔍 Assert cas_analyst workflow Pending, Decline
// 🔍 Assert cas_analyst workflow Approved, Preview the Statutory Declaration PDF
```

File: client/e2e/workflows/idir/cas_analyst.spec.ts [hunk 4]
```diff
@@ -1,26 +1,140 @@
...
+      const dashboardPage = new DashboardPOM(page);
+      const operatorsPage = new OperatorsPOM(page);
+      // 🛸 Navigate to dashboard page
+      await dashboardPage.route();
+      // 🛸 Navigates to operators tile page
+      await dashboardPage.clickOperatorsTile();
+      // 🔍 Assert that the current URL is operators
+      await operatorsPage.urlIsCorrect();
+      // 🔍 Assert `Operators` view, table and data reflect role `cas_admin`
+      await operatorsPage.viewIsCorrect(UserRole.CAS_ADMIN);
```
[the remark this answers is not in the document]
Comment (shon-button) [reply client/e2e/workflows/idir/cas_analyst.spec.ts:50]: SMH, very disappointing...got caught in copy/paste making the admin\analyst specs the same..thanks for the better set of eyes.

File: client/e2e/workflows/idir/cas_analyst.spec.ts [hunk 5]
```diff
@@ -1,26 +1,140 @@
...
+      await operationsPage.formHasExpectedUX(OperationStatus.APPROVED);
+      await operationsPage.formHasExpectedUX(OperationStatus.PENDING);
+    });
+
+    test("Test details form by workflow", async ({ page }) => {
+      const operationsPage = new OperationsPOM(page);
+      // 🛸 Navigate to operations page
+      operationsPage.route();
+      // 🔍 Assert cas_admin workflow Pending, Request Changes, Undo (Request Changes), Approve
+      await operationsPage.formHasExpectedWorkflow(
```
[the remark this answers is not in the document]
Comment (shon-button) [reply client/e2e/workflows/idir/cas_analyst.spec.ts:121]: ^

File: client/e2e/workflows/idir/cas_analyst.spec.ts [hunk 6]
```diff
@@ -1,26 +1,140 @@
...
+        UserRole.CAS_ADMIN,
+        OperationStatus.PENDING,
+        2,
+      );
+      // 🔍 Assert cas_admin workflow Approved, Preview the Statutory Declaration PDF
+      await operationsPage.formHasExpectedWorkflow(
+        UserRole.CAS_ADMIN,
+        OperationStatus.APPROVED,
+        3,
```
[the remark this answers is not in the document]
Comment (shon-button) [reply client/e2e/workflows/idir/cas_analyst.spec.ts]: The function itself needs the status as that is the driver of selecting the row, and I believe Sepehr setup the data where the Approved record is the one that actually has a download link.

File: client/e2e/workflows/idir/cas_admin.spec.ts [hunk 7]
```diff
@@ -32,82 +32,107 @@ test.afterAll(async () => {
...
-    // 🔍 Assert cas_admin workflow Existing Operator, Pending: Approve
-    await operatorsPage.formHasExpectedWorkflow(UserRole.CAS_ADMIN, 3);
-    // 🔍 Assert cas_admin workflow Existing Operator,  Pending: Reject
-    await operatorsPage.formHasExpectedWorkflow(UserRole.CAS_ADMIN, 4);
+    test("Test details view by status", async ({ page }) => {
+      const operatorsPage = new OperatorsPOM(page);
+      // 🛸 Navigate to operators page
+      operatorsPage.route();
+      // 🔍 Assert cas_admin is able to click "View Details" on see detailed info related Declined
+      await operatorsPage.formHasExpectedUX(UserOperatorStatus.DECLINED);
```
[the remark this answers is not in the document]
Comment (shon-button) [reply client/e2e/workflows/idir/cas_admin.spec.ts:74]: Appreciate your thought.
The POMs definitively abstract the logic which, on the flip side of any possible loss of self-documenting, is the gain of:
- Better code organization
- Increased reusability
- Easier collaboration

**Comment by shon-button** [comment]
@BCerki 

**New**
- Added happo screenshots
- Asserted grid column values are in expected values

File: client/e2e/workflows/idir/cas_admin.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (BCerki) [reply client/e2e/workflows/idir/cas_admin.spec.ts:99]: When you run playwright, all you can see is the test name, so I think it's helpful for it to be descriptive. But not strong feelings

File: client/e2e/workflows/idir/cas_analyst.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (BCerki) [reply client/e2e/workflows/idir/cas_analyst.spec.ts:175]: I'm still seeing some cas_admin etc in the poms

**Review by BCerki — COMMENTED** [review]
Looking good!

I'm not able to see the new happo shots in the CI step. Usually they would just show up when `happoPlaywright.screenshot` is added--has something changed in the setup, or maybe from the giraffe branch?

**Review by BCerki — COMMENTED** [review]
A couple more notes:
- I'm still seeing cas_admin in the comments in the POMs--it's a nit so fine if you don't want to change, but your comment was "yes," so maybe just missed?
- I can see the happo shots now, nice work sorting that out! I think there might be some timing issues though, because all of the shots show grids, and based on the titles, some of them are meant to be showing details pages

**Comment by shon-button** [comment]
> A couple more notes:
> 
> * I'm still seeing cas_admin in the comments in the POMs--it's a nit so fine if you don't want to change, but your comment was "yes," so maybe just missed?
> * I can see the happo shots now, nice work sorting that out! I think there might be some timing issues though, because all of the shots show grids, and based on the titles, some of them are meant to be showing details pages

@BCerki  thank-you for catching these details 🙌

**Review by BCerki — COMMENTED** [review]
Getting close! I can see all the details pages now. 

I think we've still got some flakiness/timing issues in some of the shots, however. In the latest run, I'm unable to see some of the form data (e.g. https://happo.io/a/336/p/1755/compare/5f0a5495a150fb391b898713b0747e1e4ddb4750/b89959275737bd1198a3e4b51e6e32babaf084a5?t=added&target=chrome#Operations%5C%20Details%5C%20Page%5C%20cas_admin-approved). 
I'm also not always seeing the approved/declined message at the top of the page. For example, for cas_admin, I can see declined but not approved: https://happo.io/a/336/p/1755/compare/5f0a5495a150fb391b898713b0747e1e4ddb4750/b89959275737bd1198a3e4b51e6e32babaf084a5?t=added&component=Operations%20Details%20Page%20cas_admin&target=chrome

**Pull request closed without merging**